runtime/lua/provider/python/health.lua

```diff
@@ -0,0 +1,490 @@
+local health = vim.health
+
+local iswin = vim.loop.os_uname().sysname == 'Windows_NT'
+
+local M = {}
+
+local function is(path, ty)
+  if not path then
+    return false
+  end
+  local stat = vim.loop.fs_stat(path)
+  if not stat then
+    return false
+  end
+  return stat.type == ty
+end
+
+-- Resolves Python executable path by invoking and checking `sys.executable`.
+local function python_exepath(invocation)
+  local python = vim.fn.fnameescape(invocation)
+  local out = vim.fn.system(python .. ' -c "import sys; sys.stdout.write(sys.executable)"')
+  assert(vim.v.shell_error == 0, out)
+  return vim.fs.normalize(vim.trim(out))
+end
+
+local function is_blank(v)
+  return v:find('^%s*$') ~= nil
+end
```

Comment:
This looks like it's to replace `empty()`. For empty strings, you can just compare against `""` instead.